src/core/kilocode/agent-manager/askErrorParser.ts

```diff
@@ -0,0 +1,48 @@
+import { KilocodePayload } from "./CliOutputParser"
+
+type DebugLogger = (message: string) => void
+
+export function extractPayloadMessage(
+	payload: KilocodePayload,
+	defaultMessage: string,
+	_debugLog?: DebugLogger,
+): string {
+	const rawText =
```

Comment:
Do we have zod in the extension? Seems like it would be much nicer here and throughout the rest of the PR where we parse.